core/src/main/resources/io/cucumber/core/plugin/html/formatter.js

```diff
@@ -121,7 +121,7 @@ CucumberHTML.DOMFormatter = function(rootNode) {
     if (currentStepIndex == 1) {
       this.dummyStep();
     }
-    currentStep.append('<pre class="embedded-text">' + text + '</pre>');
+    currentStep.append('<pre class="embedded-text">' + String(text).replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/"/g, '&quot;') + '</pre>');
```

Comment:
My javascript is a little bit rusty but shouldn't this be done using the `document.createTextNode(text)` instead? I realize the current implementation is using string concatenation to create the embedded text everywhere but that seems like a bad idea all together.

Reply:
I try `currentStep.append('<pre class="embedded-text">' + $(text).text() + '</pre>');`

I try `currentStep.append('<pre class="embedded-text">' + document.createTextNode(text) + '</pre>');`

but do not work.

Reply:
"we'll eventually deprecate this html formatter."  => there will be no more html report?

Reply:
> "we'll eventually deprecate this html formatter." => there will be no more html report?

I'm not sure yet how that will play out yet. The current html formatter can't handle the rule key word that was introduced into Gherkin a while ago and the current html formatter is decrepit to the point that's not worth maintaining.

We're working on a new html formatter written in node that uses react. Its currently a stand alone process, if we can turn it into single page application we'll replace the current html formatter with that.

Just wanted to let you know that trying to patch this up it may not be the best investment of your time.

Reply:
> I try currentStep.append('<pre class="embedded-text">' + $(text).text() + '</pre>');

In both cases you're trying to concatenate a string and a html node. I believe you may have to get hold of the `pre` node and append the text node to it's children.